Spec/Auth.swift

```diff
@@ -3619,5 +3619,331 @@ class Auth : QuickSpec {
+                        let token = ARTTokenDetails(token: getJWTToken(invalid: true)!)
+                        completion(token, nil)
+                    }
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+
+                    it("fails to connect") {
+                        waitUntil(timeout: testTimeout) { done in
+                            client.connection.once(.failed) { stateChange in
+                                expect(stateChange!.reason!.code).to(equal(40144))
+                                expect(stateChange!.reason!.description).to(contain("invalid signature"))
+                                done()
+                            }
+                            client.connect()
+                        }
+                    }
+                }
+            }
+
+            context("when token expires and has a means to renew") {
+
+                it("reconnects using authCallback and obtains a new token") {
+                    let tokenDuration = 3.0
+                    let options = AblyTests.clientOptions()
+                    options.useTokenAuth = true
+                    options.autoConnect = false
+                    options.authCallback = { tokenParams, completion in
+                        let token = ARTTokenDetails(token: getJWTToken(expiresIn: Int(tokenDuration))!)
+                        completion(token, nil)
+                    }
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    var originalToken = ""
+                    var originalConnectionID = ""
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.connected) { _ in
+                            originalToken = client.auth.tokenDetails!.token
+                            originalConnectionID = client.connection.id!
+
+                            client.connection.once(.disconnected) { stateChange in
+                                expect(stateChange!.reason!.code).to(equal(40142))
+
+                                client.connection.once(.connected) { _ in
+                                    expect(client.connection.id).to(equal(originalConnectionID))
+                                    expect(client.auth.tokenDetails!.token).toNot(equal(originalToken))
+                                    done()
+                                }
+                            }
+                        }
+                        client.connect()
+                    }
+                }
+            }
+            
+            context("when the token request includes a clientId") {
+                let clientId = "JWTClientId"
+                let options = AblyTests.clientOptions()
+                options.tokenDetails = ARTTokenDetails(token: getJWTToken(clientId: clientId)!)
+                options.autoConnect = false
```

Comment:
More `autoConnect` false although not sure why.

Reply:
done https://github.com/ably/ably-ios/pull/714/commits/4b5ada45f280b71afac5baf90f1e2ec638ef34f4